ui/src/app/pages/admin/user/admin-user-egress-bypass.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import { Calendar } from 'primereact/calendar';
+import { Column } from 'primereact/column';
+import { DataTable } from 'primereact/datatable';
+
+import { CreateEgressBypassWindowRequest } from 'generated/fetch';
+
+import { Button } from 'app/components/buttons';
+import { FlexColumn, FlexRow } from 'app/components/flex';
+import { TextAreaWithLengthValidationMessage } from 'app/components/inputs';
+import { TooltipTrigger } from 'app/components/popups';
+import { userAdminApi } from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import { formatDate, isDateValid } from 'app/utils/dates';
+const MIN_BYPASS_DESCRIPTION = 10;
+const MAX_BYPASS_DESCRIPTION = 4000;
+
+interface Props {
+  userId: number;
+}
+
+export const AdminUserEgressBypass = (props: Props) => {
+  const [startTime, setStartTime] = useState(null);
+  const [byPassDescription, setBypassDescription] = useState('');
+  const [apiError, setApiError] = useState(false);
+  const [bypassWindowsList, setBypasswindowsList] = useState([]);
+  const [reload, setReload] = useState(false);
+
+  useEffect(() => {
+    const { userId } = props;
+    userAdminApi()
+      .listEgressBypassWindows(userId)
+      .then((res) => setBypasswindowsList(res.bypassWindows));
+  }, [reload]);
+
+  const invalidReason =
+    !byPassDescription ||
+    byPassDescription.length < MIN_BYPASS_DESCRIPTION ||
+    byPassDescription.length > MAX_BYPASS_DESCRIPTION;
+
+  let egressBypassButtonDisabled =
+    apiError || invalidReason || !isDateValid(startTime);
+
+  const getToolTipContent = apiError ? (
+    'Error occurred while create egress bypass request'
+  ) : (
+    <div>
+      Required to lock workspace:
```

Comment:
incorrect

Reply:
Updated

Reply:
Not sure about the change, but it doesn't need to block the PR.  We can work on this later.